kratos/utilities/geometry_utilities/brep_trimming_utilities.h

```diff
@@ -77,7 +77,32 @@ namespace Kratos
         //Triangulation 
         static void Triangulate_OPT(const Clipper2Lib::Path64& polygon, std::vector<Matrix>& triangles, const double factor)
         {
-            if (polygon.size() == 4)
+            array_1d<double, 2> p1, p2, p3, p4;
+            int bestvertex;
+            double weight = 0;
+            double d1, d2 = 0.0;
+            double minweight = std::numeric_limits<double>::max();
+            Diagonal diagonal, newdiagonal;
+
+            std::list<Diagonal> diagonals;
```

Comment:
Are you sure you need an `std::list` here?

`std::list` is a linked list and thus incredibly slow and hostile to the cache. As far as I can see, your use case matches a FIFO container, in which case I recommend an [`std::queue`](https://en.cppreference.com/w/cpp/container/queue).

Reply:
Sure, `std::queue` works in this case. Thanks for your suggestion.